ambari-server/src/main/java/org/apache/ambari/server/state/Mpack.java

```diff
@@ -142,22 +151,100 @@ public void setDefinition(String definition) {
+   */
+  public RepositoryXml getRepositoryXml() {
+    return repositoryXml;
+  }
+
+  /**
+   * Gets the repository XML representation.
+   *
+   * @param repositoryXml
+   *          the {@link RepoUtil#REPOSITORY_FILE_NAME} unmarshalled.
+   */
+  public void setRepositoryXml(RepositoryXml repositoryXml) {
+    this.repositoryXml = repositoryXml;
+  }
+
+  /**
+   * Gets the module with the given name. Module names are service names.
+   *
+   * @param moduleName
+   *          the name of the module.
+   * @return the module or {@code null}.
+   */
+  public Module getModule(String moduleName) {
+    for (Module module : modules) {
+      if (StringUtils.equals(moduleName, module.getName())) {
+        return module;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Gets a component from a given module.
+   *
+   * @param moduleName
+   *          the module (service) name.
+   * @param moduleComponentName
+   *          the name of the component.
+   * @return the component or {@code null}.
+   */
+  public ModuleComponent getModuleComponent(String moduleName, String moduleComponentName) {
+    for( Module module : modules ) {
+      ModuleComponent moduleComponent = module.getModuleComponent(moduleComponentName);
+      if( null != moduleComponent ) {
+        return moduleComponent;
+      }
+    }
+
+    return null;
+  }
+
   @Override
   public boolean equals(Object o) {
-    if (this == o) return true;
-    if (o == null || getClass() != o.getClass()) return false;
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
 
     Mpack mpack = (Mpack) o;
 
-    if (!resourceId.equals(mpack.resourceId)) return false;
-    if (registryId != null ? !registryId.equals(mpack.registryId) : mpack.registryId != null) return false;
-    if (!mpackId.equals(mpack.mpackId)) return false;
-    if (!name.equals(mpack.name)) return false;
-    if (!version.equals(mpack.version)) return false;
-    if (!prerequisites.equals(mpack.prerequisites)) return false;
-    if (!modules.equals(mpack.modules)) return false;
-    if (!definition.equals(mpack.definition)) return false;
-    if (!description.equals(mpack.description)) return false;
+    if (!resourceId.equals(mpack.resourceId)) {
+      return false;
+    }
+    if (registryId != null ? !registryId.equals(mpack.registryId) : mpack.registryId != null) {
+      return false;
+    }
+    if (!mpackId.equals(mpack.mpackId)) {
+      return false;
+    }
+    if (!name.equals(mpack.name)) {
+      return false;
+    }
+    if (!version.equals(mpack.version)) {
+      return false;
+    }
+    if (!prerequisites.equals(mpack.prerequisites)) {
+      return false;
+    }
+    if (!modules.equals(mpack.modules)) {
+      return false;
+    }
+    if (!definition.equals(mpack.definition)) {
+      return false;
+    }
+    if (!description.equals(mpack.description)) {
+      return false;
+    }
```

Comment:
I'm assuming this is all an autoformat thing, you usually are the first to point out EqualsBuilder :)

Reply:
Yeah, it auto-expanded it. I try to keep incidental changes to a minimum, but I'll fix this since it's yucky :)